docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,293 @@
+cluster, or find the debugging information that is going to help me resolve the
+problem. I should be able to access a log of the deployment activity that will
+provide all the details of what happened, including both the output from the
+console as well as more verbose details from the process.
+
+As a user, I may not always be able to understand and fix the cause of a
+failure. In this case, I need this captured information so I can provide it to
+support or other community members so they can help me resolve my issue.
+
+#### Functional Requirements
+
+To address these needs, the following requirements must be met.
+
+##### FR1 - Log high level steps
+
+The output should be grouped based on the high level steps of the process. So
+rather than the current:
+
+```txt
+Doing something.
+Start creating cluster...
+Saving file.
+Installing xyz
+Waiting for x...
+```
+
+All output is at the same level and there is no differentiation or
+relevance of each log line. The new output should have clear high level steps:
+
+```txt
+☑️  Determining cluster settings
+   [specific useful details]
+☑️  Performing step 2
+   [specific useful details]
+☑️  Performing step 3
+   [specific useful details]
+```
+
+Emojis may vary to help relate to the type of step being performed - the ☑️
+above is for illustration purposes only.
+
+##### FR2 - Log relevant information for each high level step
+
+The amount of lines should be minimized. No output should be given if it does
+not directly help the user understand what they need to know.
+
+With the output grouped into the high level sections as described in FR1,
+within those high level sections there should be specific and useful details
+for the user. Output from the currently being developed `unmanaged-cluster`
+creation provides a good example of this:
+
+```txt
+🎨 Selected base image
+   projects.registry.vmware.com/tce/kind/node:v1.21.5
+
+📦 Selected core package repository
+   projects.registry.vmware.com/tkg/packages/core/repo:v1.21.2_vmware.1-tkg.1
+
+📦 Selected additional package repositories
+   projects.registry.vmware.com/tce/main:0.9.1
+
+📦 Selected kapp-controller image bundle
+   projects.registry.vmware.com/tkg/packages/core/kapp-controller:v0.23.0_vmware.1-tkg.1
+
+📧 Installing package repositories
+   Core package repo status: Reconcile succeeded
+
+🌐 Installing CNI
+   antrea.tanzu.vmware.com:0.13.3+vmware.1-tkg.1
+
+✅ Cluster created
+```
+
+Output is minimal, and can be skipped altogether if there isn't anything useful
+for the end user to know.
+
+#### FR3 - Output should be captured to a persistent log file
+
+If the users terminal closes or content scrolls off their buffer, they need to
+be able to get the details from the cluster creation. This is especially
+critical when something goes wrong and someone else needs to take a look at the
+output to help troubleshoot the failure.
+
+* Cluster creation output should be captured to
+  `~/.config/tanzu/tkg/bootstrap-logs/${CLUSTER_NAME}.log`
```

Comment:
I'm going to change this to `logs` instead of `bootstrap-logs` since I think this part will be useful for more than just bootstrap logging. For example, if a cluster is deployed, then later scaled, then upgraded, then deleted and the deletion failed... this log should capture all of that activity with the cluster.